lib/libutee/tee_api_operations.c

```diff
@@ -1418,11 +1420,6 @@ TEE_Result TEE_AEEncryptFinal(TEE_OperationHandle operation,
 		goto out;
 	}
 
-	if (!srcData && !srcLen) {
-		res = TEE_SUCCESS;
-		goto out;
-	}
-
```

Comment:
OK, I think we should take this patch in a separate PR, we'll need a test case also.